### PR TITLE
upgrade flask_login to 0.5.0

### PR DESCRIPTION
**Pull request by leohemsted**
flask_login sets a bunch of variables in the session object. We only use one of them, `user_id`. We set that to the user id from the database, and refer to it all over the placye.

However, in flask_login 0.5.0 they prefix this with an underscore to prevent people accidentally overwriting it etc. So when a user logs in we need to make sure that we set user_id manually so we can still use it.

I've manually tested inviting a new user, and the send one off flow to make sure session stuff still behaves reasonably. It seems okay but if anyone can think of anything else to test that'd be appreciated

### CODE REVIEW

File: app/models/user.py [hunk 1]
```diff
@@ -138,6 +138,7 @@ def activate(self):
             return self
 
     def login(self):
+        session['user_id'] = self.id
```
Comment (idavidmcdonald) [inline app/models/user.py]: I think this should come after the `login_user` call as if `login_user` either returns False or throws an exception, we probably don't want to be setting the user_id on the session.

https://github.com/maxcountryman/flask-login/blob/ade6bbdc521fa5e4dbed311d3e056f2e64ab4a3c/flask_login/utils.py#L145

There might not be an issue if an exception is not caught as maybe `session['user_id'] = self.id` won't actually "commit" that to the session until the end of the request or something like that.

In general, I might be worrying about something that couldn't happen in practicality but might be good to lean on the side of safety.

Oh and also all your concourse tests are breaking :(

File: tests/app/main/views/test_api_integration.py [hunk 2]
```diff
@@ -172,13 +172,13 @@ def test_api_documentation_page_should_redirect(
 
 def test_should_show_empty_api_keys_page(
     client,
-    api_user_pending,
+    api_user_active,
```
Comment (idavidmcdonald) [inline tests/app/main/views/test_api_integration.py]: What has prompted this change? Is it an unrelated fix/improvement?

Comment (leohemsted) [reply tests/app/main/views/test_api_integration.py]: the test broke following this. pending user is the state when you're mid-way through creating your account so makes no sense for a test around the api key page

File: app/models/user.py (hunk 1 above)
Comment (leohemsted) [reply app/models/user.py]: a valid concern, although after looking at the code i don't think this matters in practice for a couple of reasons.

* we only call `User.login` in two places (`log_in_user` and `activate_user`), and both of those times we activate the user immediately before hand, so login will never return false
* it'll never raise since it only raises if you do stuff with the duration parameter
* we don't use user_id for checking login state anywhere - we use the `current_user.is_authenticated` method for that, which calls through to the flask_login base class

still i dont think it hurts to change it, it's a good principle to follow on a sensitive part of the code

File: tests/__init__.py [hunk 3]
```diff
@@ -27,6 +28,9 @@ def login(self, user, mocker=None, service=None):
 
         with patch('app.events_api_client.create_event'):
             login_user(model_user)
+        with self.session_transaction() as test_session:
```
Comment (idavidmcdonald) [inline tests/__init__.py]: We are using `self.session_transaction` further up.

Is it worth calling them all `test_session` or just `session` for consistency?

Is there a reason this needs to be in a separate with statement or could it go in say the first with statement?

Comment (idavidmcdonald) [reply tests/__init__.py]: Can we also get rid of line 21 in this method? Is it still needed?

Comment (leohemsted) [reply tests/__init__.py]: this needs to be below login_user so that it gets the session variables, as i detailed in the commit message.

Comment (leohemsted) [reply tests/__init__.py]: i've called it test_session to distinguish it from flask session

Comment (leohemsted) [reply tests/__init__.py]: line 21 needs to stay because `login_user` is the flask_login function, not our own `User.login` function
